### PR TITLE
Fixing the issue with removing the commercetools API Extension pointing to other integrations

### PR DESCRIPTION
**Pull request by marcin-slezak**
- fixing the issue with removing the commercetools API Extension pointing to other integrations
    - removed CLI commands: `register` and `unregister`
    - added CLI commands: `api-extension-add`, `api-extension-update`, `api-extension-delete` and `api-extension-list`
    - added new optional configuration (`COMMERCE_TOOLS_API_EXTENSION_KEY` environment variable) to recognize own commercetools API Extension from 3rd party ones when performing delete or update operations

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,8 +21,10 @@
     "test:e2e": "jest --config ./test/jest-e2e.json",
     "prepare": "husky install",
     "config": "npm run command config",
-    "register": "nest start --config nest-commander.json -- register",
-    "unregister": "nest start --config nest-commander.json -- unregister",
+    "api-extension-list": "nest start --config nest-commander.json -- api-extension-list",
+    "api-extension-add": "nest start --config nest-commander.json -- api-extension-add",
+    "api-extension-update": "nest start --config nest-commander.json -- api-extension-update",
+    "api-extension-delete": "nest start --config nest-commander.json -- api-extension-delete",
```
Comment (jkaliszuk) [inline package.json:27]: Change `"version": "0.0.1"` to `"version": "2.0.0"` to be visible on GitHub badges.

File: README.md [hunk 2]
```diff
@@ -190,11 +197,13 @@ Currently we cover the following scenarios:
 
 - `npm run start` - start the application in production mode
 - `npm run dev` - start the application in development mode
-- `npm run register` - configure commercetools API Extension to point to your development server
-- `npm run unregister` - unregister commercetools API Extension
+- `npm run api-extension-add` - add commercetools API Extension pointing to your server (server url is taken from APP_URL environment variable)
+- `npm run api-extension-delete` - remove commercetools API Extension by Key value configured in COMMERCE_TOOLS_API_EXTENSION_KEY environment variable. Optionaly you can provide specific API Extension Id by `npm run api-extension-delete -- --id=xxx-xxx-xxx`
```
Comment (jkaliszuk) [inline README.md:201]: Maybe consider adding more information and differences between ID and ATTR in registration and unregistration actions. And not to use the same attribute multiple times may it result in the removal of all connectors with such ATTR. Unless we recognise it as obvious. As I found `COMMERCE_TOOLS_API_EXTENSION_KEY` is used as ATTR each time in actions like `UPDATE`, 'ADD' but `DELETE` could deleting ATTR or ID as well.

File: src/api-extension/api-extension.service.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            .withId({ ID: extension.id })
+            .delete({ queryArgs: { version: extension.version } })
+            .execute()
+        ).body,
+      );
+    }
+
+    return removedExtensions;
+  }
+
```
Comment (jkaliszuk) [inline src/api-extension/api-extension.service.ts:70]: Consider adding return types in `removeByAttr`, `removeById`, `list`.

File: src/api-extension/api-extension.service.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  async update(url: string): Promise<string | false> {
+    const apiExtensionKey = this.configService.get<string>(
+      'COMMERCE_TOOLS_API_EXTENSION_KEY',
+    );
+
+    if (!apiExtensionKey) {
+      throw new Error(`Missing COMMERCE_TOOLS_API_EXTENSION_KEY configuration`);
+    }
+
+    await this.removeByAttr('key', apiExtensionKey);
```
Comment (jkaliszuk) [inline src/api-extension/api-extension.service.ts:110]: Maybe check if remove was successful and then register new application with given key

**Review by jkaliszuk — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (marcin-slezak) [reply README.md:201]: By default, we are using "Key" attribute, fresh installations do not need to use the option with ID. Removing the API Extension by ID is needed only when migrating from v1x to v2 and it's described in the migration section

File: src/api-extension/api-extension.service.ts (hunk 3 above)
Comment (marcin-slezak) [reply src/api-extension/api-extension.service.ts:70]: maybe later

File: src/api-extension/api-extension.service.ts (hunk 4 above)
Comment (marcin-slezak) [reply src/api-extension/api-extension.service.ts:110]: it's intended to work that way

**Pull request merged**